### PR TITLE
Add support for bitbucket download URL

### PR DESCRIPTION
**Pull request by TG1999**
Signed-off-by: TG1999 <tushar.goel.dav@gmail.com>

### CODE REVIEW

File: src/packageurl/contrib/url2purl.py [hunk 1]
```diff
@@ -425,12 +425,30 @@ def build_bitbucket_purl(url):
...
+
+    qualifiers = {}
+    if matches:
+        qualifiers['download_url'] = url
+        return PackageURL(
+        type='bitbucket',
+        namespace=namespace,
+        name=name,
+        qualifiers=qualifiers
+    )
```
Comment (tdruez) [inline src/packageurl/contrib/url2purl.py:450]: Indenting needs to be fixed here.

**Review by tdruez — CHANGES_REQUESTED** [review]

File: src/packageurl/contrib/url2purl.py (hunk 1 above)
Comment (TG1999) [reply src/packageurl/contrib/url2purl.py:450]: Sure, changing it, should I run black over this file ??

Comment (tdruez) [reply src/packageurl/contrib/url2purl.py:450]: Probably not in the scope of this PR. 
I don't see black being used on this repo yet.
Simply fix the new code and create a ticket to suggest using black for the whole repo ;)

Comment (TG1999) [reply src/packageurl/contrib/url2purl.py:450]: Cool, will do that :)

File: src/packageurl/contrib/url2purl.py [hunk 2]
```diff
@@ -299,9 +299,8 @@ def build_cargo_purl(url):
 
 
 github_raw_content_pattern = (
-    r"^https?://raw.githubusercontent.com/"
-    r"(?P<namespace>.+)/(?P<name>.+)/(?P<version>.+)/"
-    r"(?P<subpath>.+)$"
+    r"https?://raw.githubusercontent.com/(?P<namespace>.+)/(?P<name>.+)/"
```
Comment (TG1999) [inline src/packageurl/contrib/url2purl.py:302]: @tdruez can you please have a look at the raw.githubusercontent.com URLs, I can't understand why this regex is not working

Comment (tdruez) [reply src/packageurl/contrib/url2purl.py:302]: In those cases, try to provide more context about what "is not working" so we identify the issue and get to a solution faster.

Here's some insight of my approach to figure out the issue starting with "https://raw.githubusercontent.com/LeZuse/flex-sdk/master/frameworks/projects/mx/src/mx/containers/accordionClasses/AccordionHeader.as"

```
>>> url = "https://raw.githubusercontent.com/LeZuse/flex-sdk/master/frameworks/projects/mx/src/mx/containers/accordionClasses/AccordionHeader.as"
>>> re.compile(r"https?://raw.githubusercontent.com/(?P<namespace>.+)/(?P<name>.+)/(?P<version>[^/]+)/(?P<subpath>.*)$", re.VERBOSE).match(url).groupdict()
{'namespace': 'LeZuse/flex-sdk/master/frameworks/projects/mx/src/mx', 'name': 'containers', 'version': 'accordionClasses', 'subpath': 'AccordionHeader.as'}
```

We can see that the expected namespace is not captured properly as it is allowing the `/` in the match pattern. Looking at the `(?P<version>[^/]+)` segment, the `/` is excluded from the match for the version but missing from the `namespace` and the `name`.

A better pattern syntax could be: `r"https?://raw.githubusercontent.com/(?P<namespace>[^/]+)/(?P<name>[^/]+)/(?P<version>[^/]+)/(?P<subpath>.*)$"`

```
>>> url = "https://raw.githubusercontent.com/LeZuse/flex-sdk/master/frameworks/projects/mx/src/mx/containers/accordionClasses/AccordionHeader.as"
>>> re.compile(r"https?://raw.githubusercontent.com/(?P<namespace>[^/]+)/(?P<name>[^/]+)/(?P<version>[^/]+)/(?P<subpath>.*)$", re.VERBOSE).match(url).groupdict()
{'namespace': 'LeZuse', 'name': 'flex-sdk', 'version': 'master', 'subpath': 'frameworks/projects/mx/src/mx/containers/accordionClasses/AccordionHeader.as'}
```

Note that we would need a different approach if `/` exist in the name and namespace on that repo source.

Comment (TG1999) [reply src/packageurl/contrib/url2purl.py:302]: Got it, thanks :), will be making changes soon

Comment (tdruez) [reply src/packageurl/contrib/url2purl.py:302]: See also #53 for some ideas on how to simplify route declarations. Feedback welcome.

**Comment by TG1999** [comment]
@tdruez please check this PR, I have made the required changes, and with respect to this comment.

I have not made any changes for 
```
{'https://github.com/FasterXML/woodstox/archive/woodstox-core-5.0.2.zip': 'pkg:github/fasterxml/woodstox@woodstox-core-5.0.2',
'https://github.com/adobe-fonts/source-code-pro/archive/2.030R-ro/1.050R-it.tar.gz': 'pkg:github/adobe-fonts/source-code-pro@2.030R-ro/1.050R-it',
'https://github.com/cassandra-rb/simple_uuid/archive/simple_uuid-0.3.0.zip': 'pkg:github/cassandra-rb/simple_uuid@simple_uuid-0.3.0',
'https://github.com/djberg96/sys-filesystem/archive/sys-filesystem-1.1.4.zip': 'pkg:github/djberg96/sys-filesystem@sys-filesystem-1.1.4',
'https://github.com/freedesktop/xorg-intel-gpu-tools/archive/igt-gpu-tools-1.23.tar.gz': 'pkg:github/freedesktop/xorg-intel-gpu-tools@igt-gpu-tools-1.23',
'https://github.com/grnet/synnefo/archive/synnefo/v0.12.3.zip': 'pkg:github/grnet/synnefo@synnefo/v0.12.3',
'https://github.com/n8n-io/n8n/archive/n8n@0.23.0.tar.gz': 'pkg:github/n8n-io/n8n@n8n%400.23.0',
'https://github.com/nginx/nginx/archive/branches/stable-0.7.zip': 'pkg:github/nginx/nginx@branches/stable-0.7',}
```
these URLs, as they are working as expected. I will like to work on https://github.com/package-url/packageurl-python/issues/53 in a separate PR, so please have a look and give your feedback on it, so I can make changes :)

**Review by tdruez — APPROVED** [review]

**Comment by tdruez** [comment]
@TG1999 thanks for the good work!

**Comment by TG1999** [comment]
My pleasure :)

**Pull request merged**